src/karmabot/commands/note.py

```diff
@@ -0,0 +1,113 @@
+import datetime
+import re
+from typing import Callable, Dict, Union
+
+from karmabot.db import db_session
+from karmabot.db.karma_note import KarmaNote
+from karmabot.db.karma_user import KarmaUser
+
+NOTE_CMD_PATTERN = re.compile(r"note\s(\w+)\s?(.*)")
+NOTE_COMMANDS: Dict[str, Callable] = {
+    "add": lambda text_arg, user: _add_note(text_arg, user),
+    "del": lambda text_arg, user: _del_note(text_arg, user),
+    "list": lambda text_arg, user: _list_notes(text_arg, user),
+    "default": lambda text_arg, user: f"Sorry {user.username}, your note command was not recognized. You can use {', '.join(NOTE_COMMANDS.keys())}.",
+}
+
+
+def note(**kwargs) -> Union[None, str]:
+    """Allows the user to store and retrieve simple notes.
+
+    - Syntax for adding a note: @karmabot note add <">my note<"> (note message can be in quotes)
+    - Syntax for listing notes: @karmabot note list
+    - Syntax for removing a note: @karmabote note del 1
+    """
+
+    id_arg = str(kwargs.get("user_id"))
+    text_arg = str(kwargs.get("text"))
+    user_id = id_arg.strip("<>@")
+
+    # retrieve current user
+    user = db_session.create_session().query(KarmaUser).get(user_id)
+    note_cmd = _parse_note_cmd(text_arg)
+
+    note_cmd_fnc = NOTE_COMMANDS.get(note_cmd, NOTE_COMMANDS["default"])

```

Comment:
I would take `NOTE_COMMANDS["default"]` out of the mapping and make it a separate constant